scripts/docker/CC-Devel/bin/install_thrift.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+set -e
+
+function cleanup() {
+    echo "Cleaning up Thrift temporaries."
+    if  [ -n "${THRIFT_BUILD_DIR}" ]; then
+        rm -rf "${THRIFT_BUILD_DIR}"
+    fi
+}
+
+trap cleanup EXIT
+
+function usage() {
+    echo "${0}"
+    echo "  -h  Print this usage information. Optional."
+    echo "  -t  Thrift version. Mandatory. For example '0.10.0'."
+    echo "  -p  Additional PATH components."
+}
+
+while getopts ":ht:p:" OPTION; do
+    case ${OPTION} in
+        h)
+            usage
+            exit 0
+            ;;
+        t)
+            THRIFT_VERSION="${OPTARG}"
+            ;;
+        p)
+            ADDITIONAL_PATH="${OPTARG}"
+            ;;
+        *)
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+if [[ -z "${THRIFT_VERSION}" ]]; then
+    echo "Thrift version should be defined." >&2
+    usage
+    exit 1
+fi
+
+if [[ -n "${ADDITIONAL_PATH}" ]]; then
+    export PATH="${ADDITIONAL_PATH}":"${PATH}"
+fi
+
+THRIFT_ARCHIVE_NAME="thrift-${THRIFT_VERSION}.tar.gz"
+THRIFT_BUILD_DIR="/tmp/thrift"
+THRIFT_SRC_DIR="${THRIFT_BUILD_DIR}/thrift"
+THRIFT_INSTALL_DIR="/opt/thrift"
+FAKE_JAVA_INSTALL_DIR="/usr/local/lib"
+JAVA_LIB_INSTALL_DIR="${THRIFT_INSTALL_DIR}/lib/java"
+
+mkdir -p "${THRIFT_SRC_DIR}"
+wget --no-verbose \
+  "http://xenia.sote.hu/ftp/mirrors/www.apache.org/thrift/${THRIFT_VERSION}/${THRIFT_ARCHIVE_NAME}" \
+  -O "${THRIFT_BUILD_DIR}/${THRIFT_ARCHIVE_NAME}"
+tar -xzf "${THRIFT_BUILD_DIR}/${THRIFT_ARCHIVE_NAME}" -C "${THRIFT_SRC_DIR}"   \
+  --strip-components=1
+rm "${THRIFT_BUILD_DIR}/${THRIFT_ARCHIVE_NAME}"
+
+configure_cmd=("./configure" "--prefix=${THRIFT_INSTALL_DIR}"                  \
+  "JAVA_PREFIX=${THRIFT_INSTALL_DIR}"                                          \
+  "--enable-libtool-lock" "--enable-tutorial=no" "--enable-tests=no"           \
+  "--with-libevent" "--with-zlib" "--without-nodejs" "--without-lua"           \
+  "--without-ruby" "--without-csharp" "--without-erlang" "--without-perl"      \
+  "--without-php" "--without-php_extension" "--without-dart"                   \
+  "--without-haskell" "--without-go" "--without-rs" "--without-haxe"           \
+  "--without-dotnetcore" "--without-d" "--without-qt4" "--without-qt5"         \
+  "--with-java")
+
+# Workaround. Downloading java components during java build needs it.
```

Comment:
I never had to do these things on my machine when installing, and my first thing after putting up a clean Ubuntu to a throwaway SSD was to put together CodeCompass.

Why do we need these?

Reply:
It can speed up compilation of thrift. Allows to automake not to test these items existence.